### PR TITLE
Network Console API Based on Specification

### PR DESCRIPTION
**Pull request by c-kruse**
* Adds a new collector API implementation based on api types generated from an OpenAPI specification.
* Replaces the collector backend in `pkg/flow` with a new one implemented using shared vanflow primitives in `pkg/vanflow`.
* Adds Biflow support - includes an updated set of metrics and new `/api/v1alpha1/connections` API.
* Updates the collector image to use the new `v2` branch of skupper-console

This is a partial implementation the specified collector API with sparse unit tests. Presently it is complete enough for our current v2 dev phase: a working skupper-console demo showing tcp-only traffic.

In brief, the added/updated packages are:

**main ./cmd/network-console-collector**: Sets up a collector and api server based on cli flags and runs them. Also contains `tools.go` to help pin a dependnecy on the code generation library `oapi-codegen`, and a `gen.go` with the implementation for `go generate ./cmd/network-console-collector`.

**main ./cmd/network-console-collector/codegen**: A script to generate common getters/setters on the generated api types that helped me implement things like filtering on start/endtime without a ton of per-type boilerplate. Invoked by `gen.go`.

**./cmd/network-console-collector/internal/api**: Mostly generated code for the /api/v1alpha1 api.

**./cmd/network-console-collector/internal/collector**: The heart of the collector. Collects records from remote sources, has "managers" that manage inferred records like process groups, process pairs, addresses or site clients/servers.

**./cmd/network-console-collector/internal/server**: Implements the generated server interface from `api`. Is mostly plumbing aside from the query filtering and pagination logic adapted from `pkg/flow`.

**./cmd/network-console-collector/internal/server/views**: Has mappers that translate vanflow records to api records. Most are kind of tedious, but some of the time the views are more like "materialized views" - where it queries other resource types to add context to the api that is not stored in the vanflow record. example: adding the target process name to a connector record.

## Testing/Preview

Deploy one of the example developer deployments in cmd/network-console-collector/resources/developer/ with the network-console-collector image updated to `quay.io/ckruse/network-console-collector:network-console-codegen` to a namespace with a running site.

For openshift deployments I have a gist with a manifest already pinned to that image as a shortcut:
```
kubectl apply -f https://gist.githubusercontent.com/c-kruse/6e096a5a5a114e1bedf6aa9107cea952/raw/d14c7e7d170dd74fd3a05b0e34e98f85bf5e0565/openshift.yaml
```

### CODE REVIEW

**Comment by c-kruse** [comment]
Valerio - no rush for feedback! Enjoy your holiday.

File: cmd/network-console-collector/internal/collector/collector.go [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+			case <-ctx.Done():
+				return nil
+			case event := <-c.events:
+				start := time.Now()
+				typ := event.GetTypeMeta()
+				event.ID()
+				for _, reactor := range reactors[typ] {
+					reactor(event, c.Records)
+				}
+				c.metrics.internal.flowProcessingTime.WithLabelValues(typ.String()).Observe(time.Since(start).Seconds())
```
Comment (fgiorgetti) [inline cmd/network-console-collector/internal/collector/collector.go:177]: TypeMeta.APIVersion is currently set to v1alpha1. If we eventually change versions to v2alpha1, for example,
and lets say we have two sites on the same VAN, one providing v1alpha1 and the other v2alpha1. Would that
cause metrics to be incompatible between these two sites?

Comment (c-kruse) [reply cmd/network-console-collector/internal/collector/collector.go:177]: Interesting question! I'm not sure about "compatibility" - they would become two separate time series (i.e. skupper_internal_processing_seconds{type=v1alpha1.RouterRecord} from collector _old_ and {type=v2alpha.SiteRecord} from collector _new_.) The crux, I think, would be any dashboards or monitors/alerts for a particular label value would need updated.

**Review by grs — APPROVED** [review]

**Review by nluaces — APPROVED** [review]
Looks good to me! 
In the README file about the Network Collector Deployment
```
1. Make sure [cert-manager](https://cert-manager.io/) is installed on your cluster. `kubectl get crd certificates.cert-manager.io`
```

That command didn't work for me, but this one did it: 
`kubectl apply  -f https://github.com/cert-manager/cert-manager/releases/latest/download/cert-manager.crds.yaml`

**Review by ajssmith — APPROVED** [review]

**Review by fgiorgetti — APPROVED** [review]
LGTM! Tested on minikube, crc and ocp.

**Pull request merged**